doc/external-commands.rst

```diff
@@ -0,0 +1,9 @@
+External Commands
+=================
+
+Cabal provides a system for external commands, akin to the ones used by tools like ``git`` or ``cargo``.
+
+If you execute ``cabal my-custom-command`` before failing with the message ``Error: cabal: unrecognised command: my-custom-command (try --help)``, Cabal will search the path for an executable named ``cabal-my-custom-command``.
+If found, it will execute ``cabal exec cabal-my-custom-command``, passing any remaining arguments to this external command.
```

Comment:
I would slightly reword this

```
If you execute ``cabal my-custom-command``, Cabal will search the path for an executable named ``cabal-my-custom-command`` and execute it, passing any remaining arguments to this external command. An error will be thrown in case the custom command is not found.
```